plugins/cilium-cni: cleanups around IPAM allocation and veth pair creation

tklauser opened the pull request:
See commit messages for details.

gandro reviewed (approved):
Thanks!

squeed reviewed (approved):
nice little cleanups. it's a good thing.

tklauser committed:
plugins/cilium-cni: allow to pass IPAM pool name in IPAM allocation

Preparatory refactoring for cases where we want to specifiy the IPAM
pool on CNI add. Currently this is left empty to select the default
pool.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
plugins/cilium-cni: check for IPv4 or IPv6 addresses earlier

Check whether either an IPv4 address or an IPv6 address or both were
provided by IPAM right after the IPAM allocation rather than later when
the veth pair was already set up. Also fix up the error messages.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
plugins/cilium-cni: declare variables in a more idiomatic manner

Follow commit 0e4c3b28adee ("cni: Declare variables in a more idiomatic
manner") and remove another C89 style "variable declaration block".

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
datapath/connector: rename SetupVethWithNames parameter

Rename tmpIfName to peerIfName. The interface name might not be
temporary depending on where the function's call site. Also remove
mention of the inexistent link name return value from the godoc comment.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
datapath/connector: declare variables in a more idiomatic manner

Remove yet another C89 style "variable declaration block".

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
datapath/connector: remove unused Endpoint2TempRandIfName

The function is unused since commit c3adaec9dc06 ("Remove IPVLAN code").
Also remove the now unused rand.RandomLowercaseStringWithLen function.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
datapath/connector: remove unused SetupVethRemoteNs return values

These are always 0 and are not used by the only caller.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
